### PR TITLE
JSON cutomization - 50% tested

### PR DESCRIPTION
**Pull request by rd2**
Most of the customization functionality linked to TBD JSON (as input) is complete. User-defined PSI & KHI library additions are well tested. The custom `:unit` (project-wide PSI defaults) is also well tested and working. The custom `:story`, `:space`, `:surface` & `:edge` features are integrated yet only @50% tested. A grind ... a few hours more.

### CODE REVIEW

File: lib/measures/tbd/resources/psi.rb [hunk 1]
```diff
@@ -149,6 +149,173 @@ def append(p)
   end
 end
 
+def processTBDinputs(surfaces, edges, set, io_f, schema_f)
```
Comment (macumber) [inline lib/measures/tbd/resources/psi.rb:152]: Is `io_f` an `IO` object or a file path?  It looks like a file path?  If so I'd suggest naming it path or something

File: lib/measures/tbd/resources/psi.rb [hunk 2]
```diff
@@ -149,6 +149,173 @@ def append(p)
...
+      end
+
+    else
+      # Log severe warning: enable to parse (invalid) user TBD JSON file
+    end
+  else
+    # No (optional) user-defined TBD JSON input file.
+    # In such cases, "set" must refer to a valid PSI set
+    raise "processTBDinputs: PSI set #{set}?" unless psi.set.has_key?(set)
+    raise "Set #{set} rimjoist?"     unless psi.set[set].has_key?(:rimjoist)
```
Comment (macumber) [inline lib/measures/tbd/resources/psi.rb:305]: Maybe make a method like `PSI::has_all_keys?`?

File: lib/measures/tbd/resources/psi.rb (hunk 1 above)
Comment (rd2) [reply lib/measures/tbd/resources/psi.rb:152]: Can do.

File: lib/measures/tbd/resources/psi.rb (hunk 2 above)
Comment (rd2) [reply lib/measures/tbd/resources/psi.rb:305]: Good idea.

**Pull request merged**